src/Compilers/Core/CodeAnalysisTest/Analyzers/AnalyzerFileReferenceTests.cs

```diff
@@ -248,25 +248,74 @@ public void TestFailedLoadDoesntCauseNoAnalyzersWarning()
             Assert.Equal(AnalyzerLoadFailureEventArgs.FailureErrorCode.UnableToCreateAnalyzer, errors.First().ErrorCode);
         }
 
-        // can't load a framework targeting generator, which these are in desktop
-        [ConditionalFact(typeof(CoreClrOnly))]
+        [ConditionalFact(typeof(CoreClrOnly), Reason = "Can't load a framework targeting generator, which these are in desktop")]
         public void TestLoadGenerators()
         {
             AnalyzerFileReference reference = CreateAnalyzerFileReference(Assembly.GetExecutingAssembly().Location);
-            var generators = reference.GetGenerators();
-            Assert.Equal(5, generators.Length);
+            var generators = reference.GetGeneratorsForAllLanguages();
+            Assert.Equal(10, generators.Length);
 
             var typeNames = generators.Select(g => g.GetType().FullName);
             Assert.Contains("Microsoft.CodeAnalysis.UnitTests.AnalyzerFileReferenceTests+TestGenerator", typeNames);
```

Comment:
Consider using: `AssertEx.SetEqual(new[] { "TestGenerator", "BaseGenerator", ... }, typeNames);` here and below.